File: lavanderia.pro/api/types/address.go
package types

import "time"

type Address struct {
	ID        string     `json:"id" bson:"_id,omitempty" uri:"id"`
	Client    string     `json:"client,omitempty" bson:"client,omitempty"`
	Name      string     `json:"name" bson:"name"`
	Position  Geometry   `json:"position" bson:"position"`
	Address   string     `json:"address" bson:"address"`
	Phone     string     `json:"phone" bson:"phone"`
	Extra     string     `json:"extra" bson:"extra"`
	CreatedAt *time.Time `json:"created_at,omitempty" bson:"created_at"`
	UpdatedAt *time.Time `json:"updated_at,omitempty" bson:"updated_at"`
	DeletedAt *time.Time `json:"deleted_at,omitempty" bson:"deleted_at"`
}

File: app/lib/models/address.dart
class Address {
  final String? id;
  late String? name;
  late String? address;
  late String? client;
  late String? phone;
  late String? extra;
  late String? createdAt;
  late String? updatedAt;

  Address({this.id, this.name, this.address, this.client, this.phone, this.extra, this.createdAt, this.updatedAt});

  factory Address.fromJson(Map<String, dynamic> json) {
    return Address(
      id: json['id'],
      name: json['name'],
      address: json['address'],
      client: json['client'],
      phone: json['phone'],
      extra: json['extra'],
      createdAt: json['created_at'],
      updatedAt: json['updated_at'],
    );
  }

  void setName(String nameNew) {
    name = nameNew;
  }

  void setAddress(String addressNew) {
    address = addressNew;
  }

  void setPhone(String phoneNew) {
    phone = phoneNew;
  }

  void setExtra(String extraNew) {
    extra = extraNew;
  }
}